### ISSUE TITLE
Change rendering according to Commonmark and new GitHub specification

### ISSUE THREAD

**Issue by john-cj** [dotnet/docfx#2326]
I know, DFM is based on GFM, not on Commonmark, becasue

> We chose GFM mainly because it's default language for writing on github. The learning curve will be smoothe is user is already familiar with how to write on github. (Source: https://github.com/dotnet/docfx/issues/168)

But now, GitHub itself adopted Commonmark specification, as described here: https://githubengineering.com/a-formal-spec-for-github-markdown/.

> Five years ago, we started building GitHub’s custom version of Markdown, GFM (GitHub Flavored Markdown) on top of Sundown, a parser which we specifically developed to solve some of the shortcomings of the existing Markdown parsers at the time.
>
> Today we’re hoping to improve on this situation by releasing a formal specification of the syntax for GitHub Flavored Markdown, and its corresponding reference implementation.
>
> This formal specification is based on CommonMark [...].

That's mean, something like

```
> Quote
>
> Quote continuation

> Another quote
```

in Commonmark and currently on GitHub will be rendered as 2 quotes. But in DocFX it will be rendered as one quote.

Does DFM/DocFX team plan to change rendering according to Commonmark/GitHub new specification?

**Comment by vwxyzh** [comment]
@john-cj we are working on [this](https://github.com/docascode/markdown).

**Comment by john-cj** [comment]
@vwxyzh DFM have some unique fetaures (cross-references, file inclusions): https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html. I like these features.

**Comment by xoofx** [comment]
> But I don't sure markdig will follow Commonmark/new GitHub specification.

markdig is meant to always keep 100% compatibility with CommonMark.
For GitHub specification, markdig tried to follow the behavior but until recently, their specs were not released. Now that they are partially bringing their specs in a CommonMark like format, it will be easier to follow these.

> I just want use standardized markdown, aka Commonmark, with DFM unique features.

That's what they are doing with the new [markdown engine](https://github.com/docascode/markdown) built on top of markdig. Markdig provides a full and solid implementation of the CommonMark specs, lots of builtin extensions and a pluggable architecture. So far, almost all DFM extensions have been backported as markdig extensions, without having to modify markdig. As part of this work, they have also a migration plan for converting previous DFM (not compatible with CommonMark) to a CommonMark compliant renderer.

Bottom line is that the new DFM renderer should be CommonMark compliant, faster and more solid.

**Comment by john-cj** [comment]
@xoofx Wow, it will be cool :smiley: The only bad thing is that it's too late to use it (version 0.1 alpha) :disappointed:

**Comment by xoofx** [comment]
> @xoofx Wow, it will be cool 😃 The only bad thing is that it's too late to use it (version 0.1 alpha) 😞

They have been working on it for the past months, and the migration is a significant endeavor as they have to make sure that all existing Microsoft docs will be rendered correctly with the new system (including a tool to migrate the previous doc to the new format), so it is understandable that they cannot switch like this easily. But it is likely that they will be able to start the migration in the coming months. The `0.1 alpha` is likely not really representative of the state of the actual rendering happening I believe, but they reserve enough room to make any changes necessary along the upcoming migration workload.

**Comment by john-cj** [comment]
@xoofx Thank you very much. Your explanations are excellent :smiley:

**Comment by john-cj** [comment]
Just wanted to state that there is excellent Commonmark implementation: [markdown-it](https://github.com/markdown-it/markdown-it). Bug-free and with plugin API. Probably it could be used here.
